### PR TITLE
Round state HUD - Refactor, fix res change, fix round 0 strings

### LINKED ISSUE #405: [BUG] Round state hud off-center after resolution change
### Build Info

96f57bed41f9607377521292c952d691696270d0

### Description

When the player changes the screen resolution, the round state hud which is usually centered, becomes off-centered as it retained the previous resolution's central position instead of updating.

### To Reproduce

1. Open game -> Create Server -> Go in
2. Spawn in
3. Change resolution to something smaller

### Expected behavior

Round state hud is still centered

### Actual behavior

Round state hud no longer looks centered, still retains position from previous resolution

### Operating System

- [ ] Windows
- [X] Linux

### Version/Distro

Arch Linux

### Machine's CPU

_No response_

### Machine's GPU

_No response_

### GPU's driver

_No response_

### Build's compiler

_No response_

### Additional context and Screenshots

_No response_

### PR DESCRIPTION
**Pull request by nullsystem**
## Description

* Off-center fixed by moving the variable settings over to `ApplySchemeSettings` (do note the diff looks a bit crap atm so view as the whole file)
* Also fix when supposed to not have visible strings it wasn't cleared, but now will explicitly clear them all properly and won't look stuck on map change
* Refactor
    * Common variables like stars and class are array-based
        * Refactored initialization and fetching of data
    * Reduce amount of variables holding position state
        * Removed `m_iYPos - 2` calculations which always comes down to 0 anyway
        * Renamed, changed types, removed variable same sizes of the other one

## Toolchain

- Linux GCC Distro Native Arch/GCC 14

## Linked Issues

- fixes #405

### CODE REVIEW

**Review by Rainyan — CHANGES_REQUESTED** [review]

File: mp/src/game/client/neo/ui/neo_hud_round_state.cpp [hunk 1]
```diff
@@ -46,172 +50,137 @@ CNEOHud_RoundState::CNEOHud_RoundState(const char *pElementName, vgui::Panel *pa
...
 	const bool starAutoDelete = true;
 	COMPILE_TIME_ASSERT(starAutoDelete); // If not, we need to handle that memory on dtor manually
 
-	for (auto star : stars) {
+	for (int i = 0; i < STAR__TOTAL; ++i)
+	{
+		static constexpr bool STARS_HW_FILTERED = false;
+
+		static const char *IP_STAR_NAMES[STAR__TOTAL] = {
+			"none", "alpha", "bravo", "charlie", "delta", "echo" "foxtrot"
```
[the remark this answers is not in the document]
Comment (nullsystem) [reply mp/src/game/client/neo/ui/neo_hud_round_state.cpp:62]: ooff good catch

File: mp/src/game/client/neo/ui/neo_hud_round_state.cpp [hunk 2]
```diff
@@ -46,172 +50,137 @@ CNEOHud_RoundState::CNEOHud_RoundState(const char *pElementName, vgui::Panel *pa
...
-	vgui::ImagePanel *stars[] = { starNone, starA, starB, starC, starD, starE, starF };
-
 	const bool starAutoDelete = true;
 	COMPILE_TIME_ASSERT(starAutoDelete); // If not, we need to handle that memory on dtor manually
 
-	for (auto star : stars) {
+	for (int i = 0; i < STAR__TOTAL; ++i)
+	{
+		static constexpr bool STARS_HW_FILTERED = false;
+
```
Comment (Rainyan) [inline mp/src/game/client/neo/ui/neo_hud_round_state.cpp:60]: Nit: storage duration doesn't matter

File: mp/src/game/client/neo/ui/neo_hud_round_state.cpp [hunk 3]
```diff
@@ -46,172 +50,137 @@ CNEOHud_RoundState::CNEOHud_RoundState(const char *pElementName, vgui::Panel *pa
...
+		static const char *IP_STAR_NAMES[STAR__TOTAL] = {
+			"none", "alpha", "bravo", "charlie", "delta", "echo", "foxtrot"
+		};
```
Comment (Rainyan) [inline mp/src/game/client/neo/ui/neo_hud_round_state.cpp:63]: Can be declared `static constexpr const char*`

File: mp/src/game/client/neo/ui/neo_hud_round_state.cpp [hunk 4]
```diff
@@ -46,172 +50,137 @@ CNEOHud_RoundState::CNEOHud_RoundState(const char *pElementName, vgui::Panel *pa
...
+		const char *name = IP_STAR_NAMES[i];
+		char ipName[32] = {};
+		char hudName[32] = {};
```
Comment (Rainyan) [inline mp/src/game/client/neo/ui/neo_hud_round_state.cpp:66]: Nit: redundant 0 initialization; `snprintf` will automatically append a terminating null char

File: mp/src/game/client/neo/ui/neo_hud_round_state.cpp [hunk 5]
```diff
@@ -46,172 +50,137 @@ CNEOHud_RoundState::CNEOHud_RoundState(const char *pElementName, vgui::Panel *pa
...
-
-	m_teamLogoColors[TEAM_JINRAI].color = COLOR_JINRAI;
-	m_teamLogoColors[TEAM_NSF].color = COLOR_NSF;
-
-	m_iPreviouslyActiveStar = m_iPreviouslyActiveTeam  = -1;
+	for (int i = 0; i < NEO_CLASS__ENUM_COUNT; ++i)
+	{
+		static const char *TEX_NAMES[NEO_CLASS__ENUM_COUNT] = {
+			"vgui/reconSmall", "vgui/assaultSmall", "vgui/supportSmall", "vgui/vipSmall"
+		};
```
Comment (Rainyan) [inline mp/src/game/client/neo/ui/neo_hud_round_state.cpp:84]: Nit: could be `static constexpr const`

File: mp/src/game/client/neo/ui/neo_hud_round_state.cpp [hunk 6]
```diff
@@ -46,172 +50,137 @@ CNEOHud_RoundState::CNEOHud_RoundState(const char *pElementName, vgui::Panel *pa
...
+	{
+		static const struct {
+			const char *logo;
+			const char *totalLogo;
+			Color color;
+		} TEAM_TEX_INFO[TEAM__TOTAL - TEAM_JINRAI] = {
+			{.logo = "vgui/jinrai_128tm", .totalLogo = "vgui/ts_jinrai", .color = COLOR_JINRAI},
+			{.logo = "vgui/nsf_128tm", .totalLogo = "vgui/ts_nsf", .color = COLOR_NSF},
+		};
+		const int texIdx = i - TEAM_JINRAI;
```
Comment (Rainyan) [inline mp/src/game/client/neo/ui/neo_hud_round_state.cpp:99]: Nit: decouple `TEAM_JINRAI` index by using `FIRST_GAME_TEAM` instead

**Review by Rainyan — APPROVED** [review]

**Review by AdamTadeusz — APPROVED** [review]

**Review by Rainyan — APPROVED** [review]

**Review by AdamTadeusz — APPROVED** [review]

**Pull request merged**